File: app/src/main/java/com/shortstay/pk/utils/AspectRatioCardView.java
package com.shortstay.pk.utils;

import android.content.Context;
import android.util.AttributeSet;
import android.view.ViewGroup;

import androidx.cardview.widget.CardView;

public class AspectRatioCardView extends CardView {

    public AspectRatioCardView(Context context) {
        this(context, null);
    }

    public AspectRatioCardView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public AspectRatioCardView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        float ratio = 1.2f;
        int ratioHeight = (int) (getMeasuredWidth() * ratio);
        setMeasuredDimension(getMeasuredWidth(), ratioHeight);
        ViewGroup.LayoutParams lp = getLayoutParams();
        lp.height = ratioHeight;
        setLayoutParams(lp);
    }
}

File: app/src/main/java/com/shortstay/pk/utils/AspectRatioCardView.kt
package com.shortstay.pk.utils

import android.content.Context
import android.util.AttributeSet
import androidx.cardview.widget.CardView

class AspectRatioCardView constructor(
    context: Context?,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : CardView(
    context!!, attrs, defStyleAttr
) {
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val ratio = 1.2f
        val ratioHeight = (measuredWidth * ratio).toInt()
        setMeasuredDimension(measuredWidth, ratioHeight)
        val lp = layoutParams
        lp.height = ratioHeight
        layoutParams = lp
    }
}